server/api/users/posts/[id].delete.ts

```diff
@@ -0,0 +1,70 @@
+import { Posts, Users } from "../../../dbModels";
+import middlewareFunction from "../../../utils/middlewareFunction";
+
+export default defineEventHandler(async (e) => {
+	//IF THE USER HAS EDIT OR ALL PERMISSIONS THEN THE USER CAN EDIT THE POSTS
+	const postId = e.context.params.id;
+	console.log(`DELETE /api/user/posts/${postId}`);
+
+	try {
+		const userId = await middlewareFunction(e);
+		console.log(userId);
+
+		if (e.req.headers.authentication == null) {
+			e.res.statusCode = 401;
+			return { msg: "No token, authorization denied" };
+		}
+		let post = await Posts.findById(postId);
+		if (!post) {
+			e.res.statusCode = 404;
+			return { msg: "Post not found" };
+		}
+		console.log(post.user);
+		if (post.user != userId) {
+			e.res.statusCode = 401;
+			return { msg: "User is not valid! Unauthorised" };
+		}
+		//Find the channel in users id
+		const userData = await Users.findById(userId);
+
+		let allowedChannels = userData?.channels.map((channel) => ({
+			id: channel.channelId,
+			channelPermissions: channel.channelPermissions,
+		}));
+		// console.log(allowedChannels);
+
+		let matchedChannel;
+
+		for (const i in allowedChannels) {

```

Comment:
just a thought!

if a user is given access to n channels then this is always going to take o(n) tim eto search for the channel.

So its better to create an entry in the db that the userhas access to this channel with permissions.

Reply:
Yeah. Even I was thinking about that I plan to made that change soon after the front-end is done. The logic used here is not optimal, I agree.